### PR TITLE
Add GetPartitions extended response

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Add MetadataApi and QueryApi metods wich returns
extended responce. Add GetPartitionsExtendedResponse to
PartitionsRepository, which returns extended response.
Add PrefetchPartitionsRequest and PrefetchPartitionsStatus. 

Relates-To: OLPEDGE-1902

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ */
+
+#pragma once
+
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <olp/core/porting/deprecated.h>
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h:27]: What is deprecated?

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+   * @brief Sets the billing tag for the request.
+   *
+   * @see `GetBillingTag()` for information on usage and format.
+   *
+   * @param billingTag The `BillingTag` string or `boost::none`.
+   *
+   * @return A reference to the updated `PrefetchTilesRequest` instance.
+   */
+  inline PrefetchPartitionsRequest& WithBillingTag(
+      boost::optional<std::string> billingTag) {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h:95]:
```suggestion
      boost::optional<std::string> billing_tag) {
```

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+   * @brief Sets the billing tag for the request.
+   *
+   * @see `GetBillingTag()` for information on usage and format.
+   *
+   * @param billingTag The rvalue reference to the `BillingTag` string or
+   * `boost::none`.
+   *
+   * @return A reference to the updated `PrefetchTilesRequest` instance.
+   */
+  inline PrefetchPartitionsRequest& WithBillingTag(std::string&& billingTag) {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h:110]:
```suggestion
  inline PrefetchPartitionsRequest& WithBillingTag(std::string&& billing_tag) {
```

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                        boost::optional<int64_t> version = boost::none) const {
+    std::stringstream out;
+    out << layer_id;
+    if (version) {
+      out << "@" << version.get();
+    }
+    if (GetBillingTag()) {
+      out << "$" << GetBillingTag().get();
+    }
+    return out.str();
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h:153]: This doesn't say much. LayerID, version and billing tag is probably for every request in the same client similar. Please add something from the partitions, e.g. first and last partition ID.
Also i think priority should also be included in the key.

File: olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.h [hunk 5]
```diff
@@ -91,9 +96,10 @@ class MetadataApi {
    * characters, contain only alpha/numeric ASCII characters  [A-Za-z0-9].
    * @param context A CancellationContext, which can be used to cancel request.
    *
-   * @return The Partitions response.
+   * @return  The result of this operation as an extended client::ApiResponse
+   * object with \c model::Partitions as a result.
    */
-  static PartitionsResponse GetPartitions(
+  static PartitionsExtendedResponse GetPartitions(
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/MetadataApi.h:102]: Please remove `PartitionsResponse` alias if not used anymore.

File: olp-cpp-sdk-dataservice-read/src/generated/api/QueryApi.h [hunk 6]
```diff
@@ -66,10 +71,10 @@ class QueryApi {
    * response.
    * @param context A CancellationContext instance which can be used to cancel
    * call of this method.
-   * @param The result of this operation as a client::ApiResponse object with \c
-   * model::Partitions as a result.
+   * @return  The result of this operation as an extended client::ApiResponse
+   * object with \c model::Partitions as a result.
    */
-  static PartitionsResponse GetPartitionsbyId(
+  static PartitionsExtendedResponse GetPartitionsbyId(
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/QueryApi.h:77]: Same here. `PartitionsResponse` can be removed probably.

**Review by andescu — APPROVED** [review]

**Review by mykhailo-kuchma — APPROVED** [review]

**Pull request merged**